tests/Core/EngineTest.php

```diff
@@ -72,14 +72,41 @@ public function testCrawlsStartUrls(): void
             $this->makeRequest('http://localhost:8000/test1'),
             $this->makeRequest('http://localhost:8000/test2'),
         ];
-        $run = new Run($startRequests);
+        $run = new Run($startRequests, 'DefaultSpider');
 
         $this->engine->start($run);
 
         $this->assertRouteWasCrawledTimes('/test1', 1);
         $this->assertRouteWasCrawledTimes('/test2', 1);
     }
 
+    public function testDoesntCrawlStartUrlsWithExistingRequestsInScheduler(): void
+    {
+        $startRequests = [
+            $this->makeRequest('http://localhost:8000/test1'),
+            $this->makeRequest('http://localhost:8000/test2'),
+        ];
+
+        $run = new Run($startRequests, 'DefaultSpider');
+
+        $dispatcher = new EventDispatcher();
+        $scheduler = new ArrayRequestScheduler($this->clock);
+        $scheduler->schedule($this->makeRequest('http://localhost:8000/test3'));
+        $engine = new Engine(
```

Comment:
I think it would be easier if we made `ArrayRequestScheduler` a private field on the test as well. Then we won't have to duplicate the entire engine setup just for this test. We could instead just do this

```php
$this->scheduler->schedule($this->makeRequest('http://localhost:8000/test3'));
```